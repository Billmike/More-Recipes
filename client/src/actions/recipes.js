import axios from 'axios';
import {
  ADD_RECIPE,
  GET_RECIPES,
  GET_ONE_RECIPE,
  GET_USER_RECIPES,
  ADD_FAVORITE_RECIPE,
  FETCH_FAVORITE_RECIPES,
  EDIT_RECIPE,
  REMOVE_RECIPE,
  ADD_REVIEW,
  UPVOTE_RECIPE,
  DOWNVOTE_RECIPE
} from './types';
import './toastrConfig';

export const addRecipe = recipe => ({
  type: ADD_RECIPE,
  recipe
});

export const getAllRecipes = (recipes) => {
  return {
    type: GET_RECIPES,
    recipes,
  };
};

export const getOneRecipe = (recipe) => {
  return {
    type: GET_ONE_RECIPE,
    id: recipe.id,
    recipe
  };
};

export const getUserRecipe = (userRecipe) => {
  return {
    type: GET_USER_RECIPES,
    userRecipe,
  };
};

export const addFavorites = (favoriteRecipes) => {
  return {
    type: ADD_FAVORITE_RECIPE,
    favoriteRecipes
  };
};

export const fetchFavorites = (favoriteRecipes) => {
  return {
    type: FETCH_FAVORITE_RECIPES,
    favoriteRecipes,
  };
};

export const editRecipe = (id, updates) => ({
  type: EDIT_RECIPE,
  id,
  updates,
});

export const removeRecipe = ({ id } = {}) => ({
  type: REMOVE_RECIPE,
  id
});

export const addReview = (review) => {
  return {
    type: ADD_REVIEW,
    review,
  };
};

export const upVoteRecipe = (id) => {
  return {
    type: UPVOTE_RECIPE,
    id,
  };
};

export const downVoteRecipe = (id) => {
  return {
    type: DOWNVOTE_RECIPE,
    id,
  };
};

export const startGetAllRecipes = () => {
  return (dispatch) => {
    axios.get('http://localhost:8000/api/v1/recipes')
      .then((res) => {
        dispatch(getAllRecipes(res.data.recipeData));
      })
      .catch((error) => {
        console.log(error);
      });
  };
}

export const startAddRecipe = (recipeData = {}) => {
  return (dispatch) => {
    const {
      name = '',
      description = '',
      img_url = 'no-img-here',
      category = '',
      ingredients = '',
      instructions = ''
    } = recipeData;

    const recipe = {
      name,
      description,
      img_url,
      category,
      ingredients,
      instructions,
    };
    axios.post('http://localhost:8000/api/v1/recipes', recipe)
      .then(() => {
        toastr.success('Recipe added successfully.');
        dispatch(addRecipe({
          id: recipe.id,
          ...recipe
        }));
      });
  };
};

export const startGetUserRecipes = () => {
  return (dispatch) => {
    axios.get('http://localhost:8000/api/v1/users/recipes')
      .then((res) => {
        dispatch(getUserRecipe(res.data.userRecipe));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startEditRecipe = (id, updates) => {
  return (dispatch) => {
    axios.put(`http://localhost:8000/api/v1/recipes/${id}/modify`, updates)
      .then(() => {
        toastr.success('Recipe edited successfully.');
        dispatch(editRecipe(id, updates));
      })
      .catch((error) => {
        console.log('_<_<_<_<_<__<_', error);
      });
  };
};

export const startRemoveRecipe = (id) => {
  return (dispatch) => {
    axios.delete(`http://localhost:8000/api/v1/recipes/${id}`)
      .then(() => {
        toastr.success('Recipe deleted successfully.');
        dispatch(removeRecipe(id));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startGetOneRecipe = (id) => {
  return (dispatch) => {
    axios.get(`http://localhost:8000/api/v1/recipes/${id}`)
      .then((res) => {
        dispatch(getOneRecipe(res.data.recipeData));
      })
      .catch((error) => {
        console.log(error);
      });
  };
};

export const startAddFavoriteRecipes = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/favorites`)
      .then((res) => {
        toastr.success(res.data.message);
        return dispatch(addFavorites(res.data));
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'Sorry, you cannot perform this action on your own recipe.') {
          toastr.error(err.response.data.message);
        }
      });
  };
};

export const startGetUserFavorites = (id) => {
  return (dispatch) => {
    axios.get(`http://localhost:8000/api/v1/users/${id}/favorites`)
      .then((res) => {
        dispatch(fetchFavorites(res.data));
      })
      .catch((err) => {
        console.log(err);
      });
  };
};

export const startUpvoteRecipe = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/votes/upvote`)
      .then((res) => {
        toastr.success(res.data.message);
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You already upvoted on this recipe') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};

export const startDownVoteRecipe = (id) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/votes/downvote`)
      .then((res) => {
        toastr.success(res.data.message);
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You already downvoted on this recipe') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};

export const startAddReview = (id, data) => {
  return (dispatch) => {
    axios.post(`http://localhost:8000/api/v1/recipes/${id}/reviews`, data)
      .then((res) => {
        dispatch(addReview(res.data));
      })
      .catch((err) => {
        if (err.response.data
          .message === 'You need to be logged in to perform this action.') {
          toastr.error(err.response.data.message);
        } else if (err.response.data
          .message === 'You can\'t post an empty review. Please, enter a happy review for this recipe.') {
          toastr.warning(err.response.data.message);
        }
      });
  };
};
