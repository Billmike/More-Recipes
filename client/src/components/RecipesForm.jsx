import React, { Component } from 'react';
import axios from 'axios';
import DropZone from 'react-dropzone';
import { connect } from 'react-redux';
import classNames from 'classnames';
import Loader from './Loader';
import validateRecipe from '../../../server/validators/validateRecipe';

class RecipesForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: this.props.recipe ? this.props.recipe.name : '',
      description: this.props.recipe ? this.props.recipe.description : '',
      imageUrl: this.props.recipe
        ? this.props.recipe.imageUrl
        : 'https://res.cloudinary.com/andela-nigeria/image/upload/v1519633786/salad.jpg',
      category: this.props.recipe ? this.props.recipe.category : 'Lunch',
      ingredients: this.props.recipe ? this.props.recipe.ingredients : '',
      instructions: this.props.recipe ? this.props.recipe.instructions : '',
      errors: {},
      loaded: true
    };
    this.onFormValueChange = this.onFormValueChange.bind(this);
    this.uploadToCloudinary = this.uploadToCloudinary.bind(this);
  }

  uploadToCloudinary() {
    const formData = new FormData();
    formData.append('file', this.state.imageUrl);
    formData.append('upload_preset', process.env.UPLOAD_PRESET);
    formData.append('api_key', process.env.UPLOAD_API_KEY);

    return axios.post(process.env.UPLOAD_URL, formData);
  }

  onFormValueChange(event) {
    this.setState({
      ...this.state,
      [event.target.name]: event.target.value
    });
  }

  handleDrop = (file) => {
    this.setState({ imageUrl: file[0] });
  };

  onSubmit = (event) => {
    event.preventDefault();

    if (this.isValid()) {
      this.setState({ errors: {} });
      this.setState({ loaded: false });
      this.uploadToCloudinary().then((response) => {
        const secureURL = response.data.secure_url;
        const recipeImage = this.state;
        recipeImage.imageUrl = secureURL;

        this.props.onSubmit({
          name: this.state.name,
          description: this.state.description,
          imageUrl: this.state.imageUrl,
          category: this.state.category,
          ingredients: this.state.ingredients,
          instructions: this.state.instructions
        });
      });
    }
  };

  isValid() {
    const { errors, isValid } = validateRecipe(this.state);
    if (!isValid) {
      this.setState({ errors });
    }

    return isValid;
  }

  render() {
    let dropzoneRef;
    if (!this.state.loaded) {
      return <Loader />;
    }
    const { errors } = this.state;
    return (
      <div>
        <main className="container">
          <form
            id="recipe-form-ID"
            className="formsubmit"
            onSubmit={this.onSubmit}
          >
            <div className="form-group">
              <label className="recipe-form-values" htmlFor="recipename">
                Recipe Name
              </label>
              <input
                className={classNames('form-control', {
                  'has-errors': errors.name
                })}
                id="recipename"
                aria-describedby="recipeNameHelp"
                name="name"
                value={this.state.name}
                onChange={this.onFormValueChange}
              />
              {errors.name && (
                <span className="help-block has-errors">{errors.name}</span>
              )}
              <small id="recipeNameHelp" className="form-text">
                Enter a name for your recipe
              </small>
            </div>
            <div className="form-row">
              <div className="form-group col-8">
                <label
                  className="recipe-form-values"
                  htmlFor="inputDescription"
                >
                  Description
                </label>
                <textarea
                  className={classNames('form-control', {
                    'has-errors': errors.description
                  })}
                  id="descriptionTextArea"
                  rows="3"
                  name="description"
                  value={this.state.description}
                  onChange={this.onFormValueChange}
                />
                {errors.description && (
                  <span className="help-block has-errors">
                    {errors.description}
                  </span>
                )}
                <small id="descriptionHelp" className="form-text">
                  Enter a short description for your recipe.
                </small>
              </div>
              <div className="form-group col-4 upload-btn-wrapper">
                <DropZone
                  id="dropZoneID"
                  ref={(node) => {
                    dropzoneRef = node;
                  }}
                  onDrop={this.handleDrop}
                  accept="image/*"
                  multiple={false}
                  className="dropzone"
                  disablePreview={false}
                >
                  {this.state.imageUrl.preview && (
                    <div>
                      <img
                        src={this.state.imageUrl.preview}
                        className="img-fluid fluid-image d-block"
                      />
                      <p className="dropzone-tag"> Click to upload an Image </p>
                    </div>
                  )}
                </DropZone>
                <button
                  id="dropZoneButton"
                  type="button"
                  onClick={() => {
                    dropzoneRef.open();
                  }}
                  className="drop-button"
                >
                  Click to upload image
                </button>
              </div>
            </div>
            <div className="form-group">
              <label htmlFor="category">Category</label>
              <select
                className="form-control custom-select"
                name="category"
                value={this.state.category}
                onChange={this.onFormValueChange}
                id="category"
              >
                <option value="lunch">Lunch</option>
                <option value="breakfast">Breakfast</option>
                <option value="dessert">Dessert</option>
                <option value="dinner">Dinner</option>
                <option value="snacks">Snacks</option>
              </select>
            </div>
            <div className="form-row">
              <div className="form-group col-md-6">
                <label className="recipe-form-values" htmlFor="recipename">
                  Ingredients
                </label>
                <textarea
                  className={classNames('form-control large-text', {
                    'has-errors': errors.ingredients
                  })}
                  id="recipeingredients"
                  name="ingredients"
                  aria-describedby="recipeNameHelp"
                  value={this.state.ingredients}
                  onChange={this.onFormValueChange}
                />
                {errors.ingredients && (
                  <span className="help-block has-errors">
                    {errors.ingredients}
                  </span>
                )}
                <small id="recipeNameHelp" className="form-text">
                  Enter your list of recipe ingredients.
                </small>
              </div>
              <div className="form-group col-md-6">
                <label
                  className="recipe-form-values"
                  htmlFor="inputDescription"
                >
                  Instructions
                </label>
                <textarea
                  className={classNames('form-control large-text', {
                    'has-errors': errors.instructions
                  })}
                  id="recipeDesc"
                  rows="3"
                  name="instructions"
                  value={this.state.instructions}
                  onChange={this.onFormValueChange}
                />
                {errors.instructions && (
                  <span className="help-block has-errors">
                    {errors.instructions}
                  </span>
                )}
                <small id="descriptionHelp" className="form-text">
                  Enter your step-by-step Instructions.
                </small>
              </div>
            </div>
            <div className="form-group">
              <button
                id="recipe-submit-button"
                className="btn btn-primary my-btn button-font btn-lg btn-block"
              >
                Submit Recipe
              </button>
            </div>
          </form>
        </main>
      </div>
    );
  }
}

export default RecipesForm;
