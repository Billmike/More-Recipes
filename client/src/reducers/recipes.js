const recipeDefaultState = [];

export default (state = recipeDefaultState, action) => {
  switch (action.type) {
    case 'ADD_RECIPE':
      return [
        ...state,
        action.recipe,
      ];
    default:
      return state;
  }
};
