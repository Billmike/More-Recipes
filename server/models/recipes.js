const recipes = [
  {
    id: 1,
    name: 'Banana milk shake',
    description: 'Probably the best milk shake you ever had in your entire life',
    category: 'Dessert',
    ingredients: ['Milk', 'Banana', 'Olive oil'],
    instructions: ['Blend the banana properly', 'Filter and shake'],
    upVote: 0,
    downVote: 0,
    favorite: 0,
    reviews: [
      {
        review: '',
      },
    ],
  },
  {
    id: 2,
    name: 'Banana milk shake',
    description: 'Probably the best milk shake you ever had in your entire life',
    category: 'Dessert',
    ingredients: ['Milk', 'Banana', 'Olive oil'],
    instructions: ['Blend the banana properly', 'Filter and shake'],
    upVote: 0,
    downVote: 0,
    favorite: 0,
    reviews: [
      {
        review: '',
      },
    ],
  },
  {
    id: 3,
    name: 'Banana milk shake',
    description: 'Probably the best milk shake you ever had in your entire life',
    category: 'Dessert',
    ingredients: ['Milk', 'Banana', 'Olive oil'],
    instructions: ['Blend the banana properly', 'Filter and shake'],
    upVote: 0,
    downVote: 0,
    favorite: 0,
    reviews: [
      {
        review: '',
      },
    ],
  },
];

export default recipes;
