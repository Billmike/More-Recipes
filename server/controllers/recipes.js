import recipes from '../models/recipes';
import validators from '../validators/validateinput';
import validateReview from '../validators/validateReview';

const recipesDB = [...recipes];

/**
 *Represents a Recipe class.
 */

class Recipes {
  static getRecipes(req, res) {
    if (req.query.sort == 'upVotes') {
      if (req.query.order == 'desc') {
        recipesDB.sort((recipe1, recipe2) => recipe2.upVote - recipe1.upVote);
      } else {
        recipesDB.sort((recipe1, recipe2) => recipe1.upVote - recipe2.upVote);
      }
    }
    return res.status(200).json({
      feed: { recipes: recipesDB },
    });
  }

  /**
   *
   * @param {*} req
   * @param {*} res
   */
  static createRecipe(req, res) {
    const validate = validators(req.body);
    const recipe = Object.assign({}, req.body, {
      id: recipesDB.length + 1,
      upVote: 0,
      downVote: 0,
      favorite: 0,
      reviews: [{
        review: '',
      }],
    });
    if (validate.valid) {
      recipesDB.push(recipe);
      res.status(201).send({ status: 'Successful.', feed: recipesDB[recipesDB.length - 1] });
    } else {
      res.status(400).send({ status: false, message: validate.message });
    }
  }

  /**
   *
   * @param {*} req
   * @param {*} res
   */

  static updateRecipe(req, res) {
    for (let i = 0; i <= recipesDB.length; i += 1) {
      if (recipesDB[i].id == req.params.recipeId) {
        const validate = validators(req.body);
        if (validate.valid) {
          recipesDB[i].name = req.body.name;
          recipesDB[i].description = req.body.description;
          recipesDB[i].category = req.body.category;
          recipesDB[i].ingredients = req.body.ingredients;
          recipesDB[i].instructions = req.body.instructions;

          return res.status(201).json({
            message: 'Success',
            recipes: { recipes: recipesDB[i] },
          });
        } else {
          res.status(401).send({ status: 'Update failed.', message: validate.message });
        }
      }
      return res.status(404).json({ status: false, message: 'Unable to modify recipe. Something is wrong.' });
    }
  }

  /**
   *
   * @param {*} req
   * @param {*} res
   */

  static deleteRecipe(req, res) {
    for (let i = 0; i <= recipesDB.length; i += 1) {
      if (recipesDB[i].id == req.params.recipeId) {
        recipesDB.splice(i, 1);
        return res.status(201).json({ status: 'Success', message: 'Successfully deleted recipe' });
      }
    }
  }

  /**
   *
   * @param {*} req
   * @param {*} res
   */

  static upVote(req, res) {
    for (let i = 0; i <= recipesDB.length; i += 1) {
      if (recipesDB[i].id == req.params.recipeId) {
        const counter = parseInt(req.body.upVote, 10);
        recipesDB[i].upVote += counter;
        return res.status(201).send({ status: 'Success.', message: 'Upvote successful.' });
      }
    }
  }

  /**
   *
   * @param {*} req
   * @param {*} res
   */
  static downVote(req, res) {
    for (let i = 0; i <= recipesDB.length; i += 1) {
      if (recipesDB[i].id == req.params.recipeId) {
        const counter = parseInt(req.body.downVote, 10);
        recipesDB[i].downVote += counter;
        return res.status(201).send({ status: 'Success.', message: 'Thanks for your feedback!' });
      }
    }
  }

  /**
   *
   * @param {*} req
   * @param {*} res
   */

  static reviews(req, res) {
    for (let i = 0; i <= recipesDB.length; i += 1) {
      if (recipesDB[i].id == req.params.recipeId) {
        const reviewValidate = validateReview(req.body);
        if (reviewValidate.valid) {
          recipesDB[i].reviews.push({
            review: req.body.review,
          });
          return res.status(201).send({ status: 'Success', message: 'Your review has been recorded. Our moderators would have a look at it.' });
        }
        res.status(400).send({ status: false, message: reviewValidate.message });
      }
    }
  }
}

export default Recipes;
