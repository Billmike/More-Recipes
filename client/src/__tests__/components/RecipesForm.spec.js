import React from 'react';
import { shallow } from 'enzyme';
import RecipesForm from '../../components/RecipesForm';
import recipes from '../fixtures/recipes';

test('Should render Recipes form properly', () => {
  const wrapper = shallow(<RecipesForm />);
  expect(wrapper).toMatchSnapshot();
});

test('Should render Recipes form properly with recipe data ', () => {
  const wrapper = shallow(<RecipesForm recipe={recipes[1]} />);
  expect(wrapper).toMatchSnapshot();
});

test('Should set name on input change', () => {
  const value = 'Fried rice';
  const wrapper = shallow(<RecipesForm />);
  wrapper.find('input').at(0).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('name')).toBe(value);
});

test('Should set description on input change', () => {
  const value = 'Amazing fried rice';
  const wrapper = shallow(<RecipesForm />);
  wrapper.find('input').at(1).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('description')).toBe(value);
});

test('Should set the category on input change', () => {
  const value = 'Lunch';
  const wrapper = shallow(<RecipesForm />);
  wrapper.find('select').at(0).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('category')).toBe(value);
});

test('Should set the ingredients on input change', () => {
  const value = 'Mocasin\nBlueberries\nCambodian nuts';
  const wrapper = shallow(<RecipesForm />);
  wrapper.find('textarea').at(0).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('ingredients')).toBe(value);
});

test('Should set the instructions on input change', () => {
  const value = 'Cook well for 10mins\nServe with relatives\nEnjoy!';
  const wrapper = shallow(<RecipesForm />);
  wrapper.find('textarea').at(1).simulate('change', {
    target: { value }
  });
  expect(wrapper.state('instructions')).toBe(value);
});

test('Should call onSubmit for valid form submission', () => {
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(<RecipesForm recipe={recipes[0]} onSubmit={onSubmitSpy} />);
  wrapper.find('form').simulate('submit', {
    preventDefault: () => { }
  });
  expect(wrapper.state('error')).toBe('');
  expect(onSubmitSpy).toHaveBeenLastCalledWith({
    name: recipes[0].name,
    description: recipes[0].description,
    category: recipes[0].category,
    img_url: 'no-img-here',
    ingredients: recipes[0].ingredients,
    instructions: recipes[0].instructions
  });
})
