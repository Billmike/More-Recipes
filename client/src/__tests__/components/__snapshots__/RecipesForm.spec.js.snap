// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipesForm/> Should render error for an invalid form submission 1`] = `
<div>
  <main
    className="container"
  >
    <form
      className="formsubmit"
      id="recipe-form-ID"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="recipe-form-values"
          htmlFor="recipename"
        >
          Recipe Name
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          name="name"
          onChange={[Function]}
          value=""
        />
        <small
          className="form-text"
          id="recipeNameHelp"
        >
          Enter a name for your recipe
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            className="recipe-form-values"
            htmlFor="inputDescription"
          >
            Description
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            name="description"
            onChange={[Function]}
            rows="3"
            value=""
          />
          <small
            className="form-text"
            id="descriptionHelp"
          >
            Enter a short description for your recipe.
          </small>
        </div>
        <div
          className="form-group col-4 upload-btn-wrapper"
        >
          <t
            accept="image/*"
            className="dropzone"
            disableClick={false}
            disablePreview={false}
            disabled={false}
            id="dropZoneID"
            maxSize={Infinity}
            minSize={0}
            multiple={false}
            onDrop={[Function]}
            preventDropOnDocument={true}
          />
          <button
            className="drop-button"
            id="dropZoneButton"
            onClick={[Function]}
            type="button"
          >
            Click to upload image
          </button>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="category"
        >
          Category
        </label>
        <select
          className="form-control custom-select"
          id="category"
          name="category"
          onChange={[Function]}
          value="Lunch"
        >
          <option
            value="lunch"
          >
            Lunch
          </option>
          <option
            value="breakfast"
          >
            Breakfast
          </option>
          <option
            value="dessert"
          >
            Dessert
          </option>
          <option
            value="dinner"
          >
            Dinner
          </option>
          <option
            value="snacks"
          >
            Snacks
          </option>
        </select>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-6"
        >
          <label
            className="recipe-form-values"
            htmlFor="recipename"
          >
            Ingredients
          </label>
          <textarea
            aria-describedby="recipeNameHelp"
            className="form-control large-text"
            id="recipeingredients"
            name="ingredients"
            onChange={[Function]}
            value=""
          />
          <small
            className="form-text"
            id="recipeNameHelp"
          >
            Enter your list of recipe ingredients.
          </small>
        </div>
        <div
          className="form-group col-md-6"
        >
          <label
            className="recipe-form-values"
            htmlFor="inputDescription"
          >
            Instructions
          </label>
          <textarea
            className="form-control large-text"
            id="recipeDesc"
            name="instructions"
            onChange={[Function]}
            rows="3"
            value=""
          />
          <small
            className="form-text"
            id="descriptionHelp"
          >
            Enter your step-by-step Instructions.
          </small>
        </div>
      </div>
      <div
        className="form-group"
      >
        <button
          className="btn btn-primary my-btn button-font btn-lg btn-block"
          id="recipe-submit-button"
        >
          Submit Recipe
        </button>
      </div>
    </form>
  </main>
</div>
`;

exports[`<RecipesForm/> Should render the RecipesForm with recipes 1`] = `
<div>
  <main
    className="container"
  >
    <form
      className="formsubmit"
      id="recipe-form-ID"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="recipe-form-values"
          htmlFor="recipename"
        >
          Recipe Name
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          name="name"
          onChange={[Function]}
          value="Amazing fried rice"
        />
        <small
          className="form-text"
          id="recipeNameHelp"
        >
          Enter a name for your recipe
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            className="recipe-form-values"
            htmlFor="inputDescription"
          >
            Description
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            name="description"
            onChange={[Function]}
            rows="3"
            value="A really amazing fried rice"
          />
          <small
            className="form-text"
            id="descriptionHelp"
          >
            Enter a short description for your recipe.
          </small>
        </div>
        <div
          className="form-group col-4 upload-btn-wrapper"
        >
          <t
            accept="image/*"
            className="dropzone"
            disableClick={false}
            disablePreview={false}
            disabled={false}
            id="dropZoneID"
            maxSize={Infinity}
            minSize={0}
            multiple={false}
            onDrop={[Function]}
            preventDropOnDocument={true}
          />
          <button
            className="drop-button"
            id="dropZoneButton"
            onClick={[Function]}
            type="button"
          >
            Click to upload image
          </button>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="category"
        >
          Category
        </label>
        <select
          className="form-control custom-select"
          id="category"
          name="category"
          onChange={[Function]}
          value="Lunch"
        >
          <option
            value="lunch"
          >
            Lunch
          </option>
          <option
            value="breakfast"
          >
            Breakfast
          </option>
          <option
            value="dessert"
          >
            Dessert
          </option>
          <option
            value="dinner"
          >
            Dinner
          </option>
          <option
            value="snacks"
          >
            Snacks
          </option>
        </select>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-6"
        >
          <label
            className="recipe-form-values"
            htmlFor="recipename"
          >
            Ingredients
          </label>
          <textarea
            aria-describedby="recipeNameHelp"
            className="form-control large-text"
            id="recipeingredients"
            name="ingredients"
            onChange={[Function]}
            value="Bread
Samolina
Beancake"
          />
          <small
            className="form-text"
            id="recipeNameHelp"
          >
            Enter your list of recipe ingredients.
          </small>
        </div>
        <div
          className="form-group col-md-6"
        >
          <label
            className="recipe-form-values"
            htmlFor="inputDescription"
          >
            Instructions
          </label>
          <textarea
            className="form-control large-text"
            id="recipeDesc"
            name="instructions"
            onChange={[Function]}
            rows="3"
            value="Cook it really well
Avoid mixing with water"
          />
          <small
            className="form-text"
            id="descriptionHelp"
          >
            Enter your step-by-step Instructions.
          </small>
        </div>
      </div>
      <div
        className="form-group"
      >
        <button
          className="btn btn-primary my-btn button-font btn-lg btn-block"
          id="recipe-submit-button"
        >
          Submit Recipe
        </button>
      </div>
    </form>
  </main>
</div>
`;

exports[`<RecipesForm/> Should render the recipe form correctly 1`] = `
<div>
  <main
    className="container"
  >
    <form
      className="formsubmit"
      id="recipe-form-ID"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="recipe-form-values"
          htmlFor="recipename"
        >
          Recipe Name
        </label>
        <input
          aria-describedby="recipeNameHelp"
          className="form-control"
          id="recipename"
          name="name"
          onChange={[Function]}
          value=""
        />
        <small
          className="form-text"
          id="recipeNameHelp"
        >
          Enter a name for your recipe
        </small>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-8"
        >
          <label
            className="recipe-form-values"
            htmlFor="inputDescription"
          >
            Description
          </label>
          <textarea
            className="form-control"
            id="descriptionTextArea"
            name="description"
            onChange={[Function]}
            rows="3"
            value=""
          />
          <small
            className="form-text"
            id="descriptionHelp"
          >
            Enter a short description for your recipe.
          </small>
        </div>
        <div
          className="form-group col-4 upload-btn-wrapper"
        >
          <t
            accept="image/*"
            className="dropzone"
            disableClick={false}
            disablePreview={false}
            disabled={false}
            id="dropZoneID"
            maxSize={Infinity}
            minSize={0}
            multiple={false}
            onDrop={[Function]}
            preventDropOnDocument={true}
          />
          <button
            className="drop-button"
            id="dropZoneButton"
            onClick={[Function]}
            type="button"
          >
            Click to upload image
          </button>
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="category"
        >
          Category
        </label>
        <select
          className="form-control custom-select"
          id="category"
          name="category"
          onChange={[Function]}
          value="Lunch"
        >
          <option
            value="lunch"
          >
            Lunch
          </option>
          <option
            value="breakfast"
          >
            Breakfast
          </option>
          <option
            value="dessert"
          >
            Dessert
          </option>
          <option
            value="dinner"
          >
            Dinner
          </option>
          <option
            value="snacks"
          >
            Snacks
          </option>
        </select>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-6"
        >
          <label
            className="recipe-form-values"
            htmlFor="recipename"
          >
            Ingredients
          </label>
          <textarea
            aria-describedby="recipeNameHelp"
            className="form-control large-text"
            id="recipeingredients"
            name="ingredients"
            onChange={[Function]}
            value=""
          />
          <small
            className="form-text"
            id="recipeNameHelp"
          >
            Enter your list of recipe ingredients.
          </small>
        </div>
        <div
          className="form-group col-md-6"
        >
          <label
            className="recipe-form-values"
            htmlFor="inputDescription"
          >
            Instructions
          </label>
          <textarea
            className="form-control large-text"
            id="recipeDesc"
            name="instructions"
            onChange={[Function]}
            rows="3"
            value=""
          />
          <small
            className="form-text"
            id="descriptionHelp"
          >
            Enter your step-by-step Instructions.
          </small>
        </div>
      </div>
      <div
        className="form-group"
      >
        <button
          className="btn btn-primary my-btn button-font btn-lg btn-block"
          id="recipe-submit-button"
        >
          Submit Recipe
        </button>
      </div>
    </form>
  </main>
</div>
`;
